chatlearn/synchronizer/parameter_sync.py

```diff
@@ -1062,6 +1072,59 @@ def clear_cache(self, sorted_send_actors_list=None, rank_mapping_list=None):
         self._clear_send_recv_param_names()
         self._clear_sorted_send_actors(sorted_send_actors_list)
 
+    def create_group_experts_regrouping(self, actor_groups, group_name=None, param_group="regroup_router"):
+        # Use self.actor2rank to ensure a globally unique number within a param_group.
+        actor_ranks = '_'.join([str(self.actor2rank[actor]) for actor in actor_groups])
+        # Always include self.group_name to ensure the name of a param_group is unique.
+        if group_name is None:
+            group_name = self.group_name
+        elif not group_name.startswith(self.group_name + "_"):
+            group_name = self.group_name + "_" + group_name
+        finalized_group_name = f"{group_name}_{param_group}_among_{actor_ranks}"
+        logger.debug(f"finalized_group_name is {finalized_group_name}")
+        logger.debug(f"current collevtive_groups is {self.collective_groups}")
+        if finalized_group_name not in self.collective_groups:
+            refs = []
+            for rank, actor in enumerate(actor_groups):
+                ref = actor.setup_collective_group.remote(rank, len(actor_groups), "nccl", finalized_group_name)
+                refs.append(ref)
+            future.wait(refs)
+            self.collective_groups.append(finalized_group_name)
+        return actor_groups, finalized_group_name
+
+    def sync_regroup_experts(self, actors, group_name, requires_grad=None, filter_fn=None, param_group="default"):
+        for actor in actors:
+            self.set_sync_param_names(actor, actor, requires_grad, filter_fn, param_group, is_parameter_changed=False)
+        pipe_stage = self.get_actor_pipe_rank(actors[0])
+        refs = []
+        for actor in actors:
+            ref = actor.allgather_expert_parameter.remote(group_name, pipe_stage)
+            refs.append(ref)
+        future.wait(refs, return_output=True)
+
+    def _multi_thread_sync_for_tp_num_mapping_eq_1(
```

Comment:
a different name would be helpful to distinguish `_multi_thread_sync_for_tp_num_mapping_eq_1` from megatron-megatron and megatron-vllm synchronization pairs. Otherwise, megatron-megatron synchronization might call this function and break execution.

Reply:
arg 'is_parameter_changed' of 'megatron2megatron ' is False.  megatron-megatron synchronization calls `_multi_thread_sync_for_tp_num_mapping_eq_1` in super class directly.